### ISSUE TITLE
DEPRECATION WARNING: The behavior of `changed?` in OrderInventory

### ISSUE THREAD

**Issue by damianlegawiec** [spree/spree#8271]
When running specs on CI:
```
DEPRECATION WARNING: The behavior of `changed?` inside of after callbacks will be changing in the next version of Rails. The new return value will reflect the behavior of calling the method after `save` returned (e.g. the opposite of what it returns now). To maintain the current behavior, use `saved_changes?` instead. (called from verify at /home/ubuntu/spree/core/app/models/spree/order_inventory.rb:28)
DEPRECATION WARNING: The behavior of `changed_attributes` inside of after callbacks will be changing in the next version of Rails. The new return value will reflect the behavior of calling the method after `save` returned (e.g. the opposite of what it returns now). To maintain the current behavior, use `saved_changes.transform_values(&:first)` instead. (called from verify at /home/ubuntu/spree/core/app/models/spree/order_inventory.rb:28)
```

**Comment by bbonislawski** [comment]
Method `verify` from OrderInventory is called from `after_save` and `before_destroy` callbacks in Spree::LineItem. 
Behavior of `changed?` is changed only in `after_save` callbacks so changing you cannot simply replace `changed?` with `saved_changes?` because it breaks `before_destroy` callback.

**Comment by bbonislawski** [comment]
Fixed with #8506
